### PR TITLE
added comments and minor edits

### PR DESCRIPTION
**Pull request by agladstein**
I've left a number of comments in the .tex

### CODE REVIEW

File: popSimManu.tex [hunk 1]
```diff
@@ -145,15 +145,24 @@ \section*{Introduction}
...
+%ALG: I think breaking up here is fine.
 
 Assessing accuracy in population genetics is challenging in large part because
  ``ground-truth'' in population genetics comes from simulation,
 not from empirical observations, and as a field there has been no agreement
-on community standards or best practices. The general modus operandi to date has been to
-introduce methods that are validated by a set of bespoke simulations produced by individual
-groups. Beyond inefficiency from duplicated effort, such a situation leads
+on community standards or best practices. 
+%ALG: Can we add an explination of why simulations are used as ground-truth?
```
Comment (andrewkern) [inline popSimManu.tex]: what if we added a phrase like "...not from empirical observations, as historical processes cannot be observed directly, and as a..."?

File: popSimManu.tex [hunk 2]
```diff
@@ -163,10 +172,15 @@ \section*{Introduction}
 %ADK: made some edits here
 %RNG: Removed the sap. :-P Feel free to put it back if you like it better.
 %JK: not crazy about it either. Making it shorter helps though?
+%ALG: I think it sounds fine.
+%ALG: Do we also need to say something about how it’s useful to people not in methods development? 
```
Comment (andrewkern) [inline popSimManu.tex]: i think a lot of these comments are dated at this point... and @asiepel is about to go a bit of an edit right here

File: popSimManu.tex [hunk 3]
```diff
@@ -295,10 +314,12 @@ \section*{Results}
...
 \stdpopsim uses the  \texttt{msprime} coalescent simulator \citep{kelleher2016efficient}
-as the default simulation engine. We plan to include other simulation
-engines to support scenarios that cannot be modelled under the coalescent framework,
-and have already implemented \texttt{SLiM} \citep{haller2019tree,haller2019slim} as
-an additional backend.
+as the default simulation engine. 
+We have implemented \texttt{SLiM} \citep{haller2019tree,haller2019slim} 
+as an alternative back-end, and plan to include other simulation
+engines to support scenarios that cannot be modelled under the coalescent framework.
+%ALG: previous version of the last sentence sounded kind of weird to me. But maybe
```
Comment (andrewkern) [inline popSimManu.tex]: this is a good suggestion

File: popSimManu.tex [hunk 4]
```diff
@@ -391,13 +412,14 @@ \subsection*{Use case: comparing methods of demographic inference}
 available from \url{https://github.com/popsim-consortium/analysis}.
 These workflows allow efficient computing in multicore or cluster environments.
 
+%ALG: When talking about the Snakemake workflow, consistently use the word “workflow”, not “pipeline”. 
```
Comment (andrewkern) [inline popSimManu.tex]: yup agreed

File: popSimManu.tex [hunk 5]
```diff
@@ -423,13 +445,15 @@ \subsection*{Use case: comparing methods of demographic inference}
...
 % JK here --- I think it would really help if we described what the models
 % were, briefly, rather than saying who inferred them.
+% ALG: agreed.
 In Figure \ref{fig:n_t_ragsdale} we present results from our $N(t)$ analysis pipeline
-run on whole genome simulations under under a sophisticated demographic model
-and empirical genetic map and a model of human migration out of Africa
-that includes archaic admiture \citep{ragsdale2019models}. In each column of this figure
+run on human whole genome simulations under a sophisticated demographic model
+and empirical genetic map and a model of migration out of Africa
+that includes archaic admiture \citep{ragsdale2019models}. In each row of this figure
```
Comment (andrewkern) [inline popSimManu.tex]: I think we should just say "OutOfAfricaArchaicAdmixture\_5R19" at this point

File: popSimManu.tex [hunk 6]
```diff
@@ -423,13 +445,15 @@ \subsection*{Use case: comparing methods of demographic inference}
...
 we show inferred $N(t)$ for the three extant populations in the model.
-In each row we show comparisons among the methods (note that we compare two differing
+In each columns we show comparisons among the methods (note that we compare two differing
 sample sizes for \MSMC).
+% ALG: I think column and row were reversed in the text description. Or am I crazy?
```
Comment (andrewkern) [inline popSimManu.tex]: no the way it is written currently is correct...

File: popSimManu.tex [hunk 7]
```diff
@@ -456,7 +480,7 @@ \subsection*{Use case: comparing methods of demographic inference}
 we can simulate under a human genome architecture and genetic map, but without demography.
 We show results of such an
 experiment in Figure \ref{fig:n_t_HomSap_constant}. Again we see that all methods
-are recovering a modern population size within a factor of two of the truth, however
+recover a modern population size within a factor of two of the truth, however
```
Comment (andrewkern) [inline popSimManu.tex:456]: 👍

**Comment by asiepel** [comment]
yes

On Dec 19, 2019, at 11:29 PM, Andrew Kern <notifications@github.com<mailto:notifications@github.com>> wrote:


@andrewkern commented on this pull request.

________________________________

In popSimManu.tex<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_popsim-2Dconsortium_manuscript_pull_112-23discussion-5Fr360223918&d=DwMFaQ&c=mkpgQs82XaCKIwNV8b32dmVOmERqJe4bBOtF0CetP9Y&r=rNAqNCgksGfltoLkih1npA&m=u1xcaIvQ6npDJVZLt6UaKWSi-uvhHXwraXumLB50zfs&s=JQ-PS_Ll6cuPW2vMejYM-LcEg2wFIgZ5iRqCLt4Cn9c&e=>:

> @@ -163,10 +172,15 @@ \section*{Introduction}
 %ADK: made some edits here
 %RNG: Removed the sap. :-P Feel free to put it back if you like it better.
 %JK: not crazy about it either. Making it shorter helps though?
+%ALG: I think it sounds fine.
+%ALG: Do we also need to say something about how it’s useful to people not in methods development?


i think a lot of these comments are dated at this point... and @asiepel<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_asiepel&d=DwMFaQ&c=mkpgQs82XaCKIwNV8b32dmVOmERqJe4bBOtF0CetP9Y&r=rNAqNCgksGfltoLkih1npA&m=u1xcaIvQ6npDJVZLt6UaKWSi-uvhHXwraXumLB50zfs&s=8wDa6G48j4Zy20CcV2nNmGXYslpwrJRdudPmb7r8YwU&e=> is about to go a bit of an edit right here

File: popSimManu.tex [hunk 8]
```diff
@@ -514,7 +538,8 @@ \subsection*{Use case: comparing methods of demographic inference}
 For simplicity, we conducted inference in \dadi and \fastsimcoal by fitting an IM model
 with constant population sizes and bi-directional migration \citep{hey2004im}. Our motivation for using
 an IM model was to mimic the approach often used on empirical datasets, where a relatively
-simple model is fit that may not reflect the true underlying demography.
+simple model that may not reflect the true underlying demography is fit.
+% ALG: Sounds better to me like that - but if someone doesn't agree grammatically, put it back.
```
Comment (andrewkern) [inline popSimManu.tex]: i prefer the way it is written currently

**Comment by asiepel** [comment]
I can try to work something like that in here….. this overlaps with text I am editing now anyway

On Dec 19, 2019, at 11:28 PM, Andrew Kern <notifications@github.com<mailto:notifications@github.com>> wrote:


@andrewkern commented on this pull request.

________________________________

In popSimManu.tex<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_popsim-2Dconsortium_manuscript_pull_112-23discussion-5Fr360223757&d=DwMCaQ&c=mkpgQs82XaCKIwNV8b32dmVOmERqJe4bBOtF0CetP9Y&r=rNAqNCgksGfltoLkih1npA&m=bsrgswP6UreCUZkBF_SBhayc5aZ8i6Irm7K2IKhw5cw&s=PObcSpWvIQVtrQDbeX3dzXFQifB_zYrEQJ9EaHpntrY&e=>:

>
 Assessing accuracy in population genetics is challenging in large part because
  ``ground-truth'' in population genetics comes from simulation,
 not from empirical observations, and as a field there has been no agreement
-on community standards or best practices. The general modus operandi to date has been to
-introduce methods that are validated by a set of bespoke simulations produced by individual
-groups. Beyond inefficiency from duplicated effort, such a situation leads
+on community standards or best practices.
+%ALG: Can we add an explination of why simulations are used as ground-truth?


what if we added a phrase like "...not from empirical observations, as historical processes cannot be observed directly, and as a..."?

**Comment by andrewkern** [comment]
awesome

File: popSimManu.tex [hunk 9]
```diff
@@ -532,7 +557,8 @@ \subsection*{Use case: comparing methods of demographic inference}
 simulated YRI population size history. Moreover, these approaches allow for estimating
 migration rates (Figure \ref{fig:homsap_mig_rates}), also leading to more accurate inference
 of divergence times. By contrast, \smcpp is much better at capturing the recent exponential
-growth in the CEU population, though the inferred population sizes are in general more
+growth in the CEU population, though the inferred population sizes are generally more
```
Comment (andrewkern) [inline popSimManu.tex]: 👍

File: popSimManu.tex [hunk 10]
```diff
@@ -667,14 +704,20 @@ \subsection*{Model QC procedures}
...
 demographic model for the relevant organism along with clear documentation
-of the model parameters and populations. This model is submitted as a Pull Request,
+of the model parameters and populations. 
+% ALG: I’m not sure what documentation of populations means. 
+This model is submitted on GitHub as a Pull Request,
 where it is evaluated by a reviewer and then included as `preliminary',
 but is not linked to the online documentation nor the command line interface.
-Inclusion of a preliminary model triggers a QC issue, and a second developer,
+Inclusion of a preliminary model 
+% ALG: automatically?
```
Comment (andrewkern) [inline popSimManu.tex:679]: actually it's not automatic

File: popSimManu.tex [hunk 11]
```diff
@@ -667,14 +704,20 @@ \subsection*{Model QC procedures}
...
+triggers a QC issue, and a second developer,
 Developer B, then independently reimplements the model from the relevant
 primary sources and adds an automatic unit test for equality between the
-QC implementation and the preliminary production model. If the two
-implementations are equal the original model is included in \stdpopsim.
+QC implementation and the preliminary production model. 
+% ALG: How is equal defined?
```
Comment (andrewkern) [inline popSimManu.tex]: see the API 😛

File: popSimManu.tex [hunk 12]
```diff
@@ -667,14 +704,20 @@ \subsection*{Model QC procedures}
...
+If the two implementations are equal the original model is included in \stdpopsim.
+% ALG: what does included in stdpopsim mean? made available in the next release and appears in catalog? 
```
Comment (andrewkern) [inline popSimManu.tex]: it means that it gets included in the documentation and appears in the catalog

File: popSimManu.tex [hunk 13]
```diff
@@ -764,11 +807,12 @@ \subsection*{Workflow for analysis of simulated data}
 model is often misspecified. To ground-truth our inference approach, we also conducted
 inference on a generic IM model that was identical to the model used for inference \ref{fig:generic_IM}.
 
-From HomSap we ran replicate whole genome simulations under the model
+For HomSap we ran replicate whole genome simulations under the model
```
Comment (andrewkern) [inline popSimManu.tex]: good catch

**Review by andrewkern — CHANGES_REQUESTED** [review]
okay @agladstein lots of comments here. Can you please keep in the changes that I have marked affirmatively and remove the others for now?

**Comment by agladstein** [comment]
So, should I rebase first?

File: popSimManu.tex [hunk 14]
```diff
@@ -145,15 +145,24 @@ \section*{Introduction}
 benchmark the quality of inferences gleaned from computational population genetics.
 As large databases of population genetic variation begin to be used to inform public health procedures,
 the accuracy and quality of these inferences is becoming ever more important.
+%ALG: cite large databases of population genetic variate to inform public health procedures? UK biobank, 23andMe???
```
Comment (petrelharp) [inline popSimManu.tex]: I think this is well-known enough so that we don't need to clutter the text with a list of citations. But if you know a good review of the topic (prospectus on future PRS score use?) then that'd be a useful thing to cite here?

File: popSimManu.tex [hunk 15]
```diff
@@ -145,15 +145,24 @@ \section*{Introduction}
...
 %CK: this first paragraph is super long - might be helpful to break up
+%ALG: I think breaking up here is fine.
```
Comment (petrelharp) [inline popSimManu.tex]: It's OK now - the only reason it's long is the extensive citation list (which I don't love but am leaving in...).

**Comment by asiepel** [comment]
numerical citations would help a lot here…. but I realize the names are nice to have elsewhere.

File: popSimManu.tex (hunk 1 above)
Comment (petrelharp) [reply popSimManu.tex]: Currently says
```
Assessing the accuracy of inference methods for population genetics is
challenging in large part because the ``ground-truth'' in question
generally comes not from direct empirical observations, as the relevant
historical processes can rarely be observed, but instead from simulations.
Population genetic simulations are therefore critically important to the
field, yet there has been no systematic attempt to establish community
standards or best practices for executing them. Instead, the general modus
```

File: popSimManu.tex [hunk 16]
```diff
@@ -145,15 +145,24 @@ \section*{Introduction}
...
+%ALG: Can we add an explination of why simulations are used as ground-truth?
+% Maybe something like: "The major goal of population genetics research is to infer past processes.
+% However, since we do not have a time machine to find out the true history,
+% the field of population genetics uses simulations to create ground-truth datasets.
+% As of yet, as a field there has..."
+The general modus operandi to date has been to
+introduce methods that are validated by a set of bespoke simulations produced by the authors. 
+Beyond inefficiency from duplicated effort, such a situation leads
 to decreased reproducibility and transparency across the entire field.
+% ALG: I like this sentence. Don't change!
```
Comment (petrelharp) [inline popSimManu.tex]: Several edits later, this has become
```
Moreover, this results in a great deal of duplicated effort,
and contributes to decreased reproducibility and transparency across the entire field.
```
... hopefully close enough?

**Comment by asiepel** [comment]
yes this was an attempt to address ALG’s comment (adapting language suggested by Andy)

File: popSimManu.tex [hunk 17]
```diff
@@ -163,10 +172,15 @@ \section*{Introduction}
...
 %JK: not crazy about it either. Making it shorter helps though?
+%ALG: I think it sounds fine.
+%ALG: Do we also need to say something about how it’s useful to people not in methods development? 
 
 It has been challenging to assess the degree to which modeling assumptions
 and choices of data summaries can affect the ultimate results of population
-genetic inference. Yet there
+genetic inference. 
+%ALG: Do we want to say here that it is expected to differ based on modeling assumptions 
+% and stats and why it is expected, before giving an example?
```
Comment (petrelharp) [inline popSimManu.tex]: Hm - yes, kinda - but that seems like a bit much to explain here.

File: popSimManu.tex [hunk 18]
```diff
@@ -182,6 +196,7 @@ \section*{Introduction}
 guidance will only increase as researchers continue to generate more genomic
 variation data from non-model taxa and wish to make inferences regarding
 demography and selection.
+% ALG: I like the last two sentences.
```
Comment (petrelharp) [inline popSimManu.tex]: still there. =)

File: popSimManu.tex [hunk 19]
```diff
@@ -197,6 +212,7 @@ \section*{Introduction}
 % inferred scenarios independently, which makes good benchmarking
 % a major additional time investment, and can lead to errors.
 % Lastly, the degree to which different methods are compared can be quite variable.
+% ALG: I agree, this can be removed.
```
Comment (petrelharp) [inline popSimManu.tex]: it's gone

File: popSimManu.tex [hunk 20]
```diff
@@ -252,6 +270,7 @@ \section*{Introduction}
 \label{fig:cartoon}
 \end{center}
 \end{figure}
+% ALG: I don’t see default generation time and mutation rate in the fig 
```
Comment (petrelharp) [inline popSimManu.tex]: true - but that'd be hard to fit in there? I'm OK with it being omitted.

File: popSimManu.tex (hunk 3 above)
Comment (petrelharp) [reply popSimManu.tex]: Changed to
```
We have implemented \texttt{SLiM} \citep{haller2019tree,haller2019slim}
as an alternative backend,
to allow simulation of processes that cannot be modeled under the coalescent.
```
as I don't think we actually have plans for non-SLiM backends?

File: popSimManu.tex (hunk 4 above)
Comment (petrelharp) [reply popSimManu.tex]: done

File: popSimManu.tex (hunk 14 above)
Comment (agladstein) [reply popSimManu.tex]: that's fine. Not off the top of my head!

File: popSimManu.tex (hunk 8 above)
Comment (petrelharp) [reply popSimManu.tex]: awkward but ok

File: popSimManu.tex [hunk 21]
```diff
@@ -532,7 +557,8 @@ \subsection*{Use case: comparing methods of demographic inference}
 simulated YRI population size history. Moreover, these approaches allow for estimating
 migration rates (Figure \ref{fig:homsap_mig_rates}), also leading to more accurate inference
 of divergence times. By contrast, \smcpp is much better at capturing the recent exponential
-growth in the CEU population, though the inferred population sizes are in general more
+growth in the CEU population, though the inferred population sizes are generally more
+% ALG: Again, sounds better to me like that - but if someone doesn't agree grammatically, put it back.
```
Comment (petrelharp) [inline popSimManu.tex]: "generally noisier"

File: popSimManu.tex [hunk 22]
```diff
@@ -557,10 +583,11 @@ \section*{Discussion}
 
 Here we have described the first major product from the PopSim Consortium:
 the \stdpopsim library. We have founded the Consortium with a number of specific goals in mind:
-standardization of simulation within the computational genetics community,
+standardization of simulation within the population genetics community,
```
Comment (petrelharp) [inline popSimManu.tex:558]: oh good catch - we are totally not working on alignment or protein structure prediction

File: popSimManu.tex [hunk 23]
```diff
@@ -557,10 +583,11 @@ \section*{Discussion}
...
 increasing reproducibility and ease of use of complex simulation based results,
 community-based development and decision making guiding best practices in population genetics,
 and eventual benchmarking of inferential methods.
+% ALG: Should these goals be mentioned in the intro?
```
Comment (petrelharp) [inline popSimManu.tex]: I think it does in other words, and we don't want to completely repeat ourselves.

File: popSimManu.tex (hunk 20 above)
Comment (agladstein) [reply popSimManu.tex]: Looks good without it.

File: popSimManu.tex [hunk 24]
```diff
@@ -574,13 +601,15 @@ \section*{Discussion}
 % All of the important points are made below (even with some of the same
 % phrases, such as "barriers to entry") and I think it is needlessly
 % redundant.
+% ALG: I'm not sure about that. I don't really see that it's covered below.
```
Comment (petrelharp) [inline popSimManu.tex:575]: I think it is covered elsewhere; the question is whether we want to emphasize it. I felt like emphasizing it, especially given the work we've put in to the API, but am leaving this commented out.

File: popSimManu.tex [hunk 25]
```diff
@@ -574,13 +601,15 @@ \section*{Discussion}
...
 
 %Individual research groups are often in the position of having to independently
 %reimplement complex, previously published demographic models.
 %Additional layers of realism (e.g., recombination maps) make this especially daunting.
 %\stdpopsim aims to lower this barrier
 %by creating a quality controlled, easy-to-use, gateway to models and organisms
 %that are central to modern research in genetics.
+% ALG: I was going to suggest adding a sentence or two here about how we will add in more models.
```
Comment (petrelharp) [inline popSimManu.tex]: we say that above, so... meh?

File: popSimManu.tex [hunk 26]
```diff
@@ -599,7 +628,12 @@ \section*{Discussion}
 (Figure~\ref{fig:IM_popn_human}), most methods do reasonably well at
 reconstructing the YRI history, but struggle with the more complex CEU
 history, in large part because of the restriction of constant sizes per
-population.  An exception is \smcpp, which does not have the same
+population.  
+% ALG: I would rephrase these to make clear it’s the simulated history.
+% Otherwise, it sounds like you are saying that CEU have a more complicated 
+% history than YRI… which I doubt is true, it’s just that in our models they 
+% have a more complicated history. 
```
Comment (petrelharp) [inline popSimManu.tex]: I am restraining myself from saying "drastically oversimplified YRI history", since it's YRI that really has the more complex history... =)

Comment (agladstein) [reply popSimManu.tex]: yeah...

Comment (petrelharp) [reply popSimManu.tex]: I've added "simulated" in front of YRI and CEU.

File: popSimManu.tex (hunk 1 above)
Comment (agladstein) [reply popSimManu.tex]: I think that conveys what I wanted better.

File: popSimManu.tex (hunk 17 above)
Comment (agladstein) [reply popSimManu.tex]: that's fine. maybe keep in mind for later.

**Comment by agladstein** [comment]
Ok, I just rebased... now I will incorporate the edits from @andrewkern and @petrelharp

File: popSimManu.tex (hunk 4 above)
Comment (agladstein) [reply popSimManu.tex]: @petrelharp are you making these changes? Because I will push with the approved changes.

File: popSimManu.tex (hunk 6 above)
Comment (agladstein) [reply popSimManu.tex]: I see. I interpreted that sentence as: Each row is a different method, and each column is a different population. I'll leave as it was, but I think the sentence is kinda confusing.

File: popSimManu.tex (hunk 24 above)
Comment (agladstein) [reply popSimManu.tex:575]: Hmm... I'd have to give it another full read through.

File: popSimManu.tex (hunk 11 above)
Comment (petrelharp) [reply popSimManu.tex]: changed to "equivalent"

File: popSimManu.tex (hunk 12 above)
Comment (petrelharp) [reply popSimManu.tex]: leaving as-is

File: popSimManu.tex (hunk 10 above)
Comment (agladstein) [reply popSimManu.tex:679]: I feel like "triggers" implies automatic...
Maybe rephrase to active. Like "Developer A creates a QC issue"

File: popSimManu.tex (hunk 11 above)
Comment (agladstein) [reply popSimManu.tex]: fair enough.

File: popSimManu.tex (hunk 12 above)
Comment (agladstein) [reply popSimManu.tex]: may want to add that into the text, but also maybe okay to leave as is.

**Comment by andrewkern** [comment]
seems like we should wait for https://github.com/popsim-consortium/manuscript/pull/127 to wrap up

**Comment by andrewkern** [comment]
@petrelharp what do you think?

**Comment by agladstein** [comment]
I'm ready to rebase to clean up my commits. Or I can wait for #127 and rebase. I'm sure there will be some conflicts.

**Comment by petrelharp** [comment]
Hm - sorry, i should have explained - rather than merging in all those comments, I went through and did something about everything you mentioned, including taking your edits. Have you added more things?

**Comment by petrelharp** [comment]
So, I think this was a useful way to provide comments but unless you've done new things, we can drop this PR.

**Comment by petrelharp** [comment]
Please do comment on suitability of my edits, and/or suggest changes, of course! But probably best to do that in a new PR.

**Comment by petrelharp** [comment]
@adkern wrote:
> Can you please keep in the changes that I have marked affirmatively and remove the others for now?

Oh dear, I just saw this; apologies for mixed messages...

**Comment by agladstein** [comment]
All I did was get up to date and incorporate the comments that @andrewkern and @petrelharp approved. If you also have included the edits in your PR #127, we can drop this one. There shouldn't be anything new from yours. I can check your branch locally too.

**Comment by agladstein** [comment]
Yeah, probably at this point, easiest to drop this PR and I can go through yours.

**Comment by petrelharp** [comment]
Sorry about that.

**Comment by agladstein** [comment]
that's okay!
I'm reviewing your PR now.
